### ISSUE TITLE
Correspondance Map & Visualization

### ISSUE THREAD

**Issue by Mayur28** [hhj1897/face_pose_augmentation#13]
Hi,

I just been working through your implementation and I am having difficulty understanding what does `result["correspondence_map"] `represent please? I noted that it has the same spatial dimensions as the input image, however, this result has 2 channels (and displaying each channel independently does not offer any insight).

Also as a side note, I just wanted to find out how can I generate visualizations of the 3D mesh as presented in the paper please (such as the visualization shown below). NB, I have also referred to the paper _High-fidelity pose and expression normalization for face recognition in the wild_ which uses these visualizations as well, however I was unable to produce these visualizations.
![image](https://user-images.githubusercontent.com/26574827/203302828-95fbf943-3168-4906-8b4a-756390e0330f.png)

Any assistance would be highly appreciated.

**Comment by hhj1897** [comment]
The correspondence map shows the per-pixel correspondence between the input image and the warped output image (see cv2.remap or torch.nn.functional.grid_sample for more details). As the rotated face is produced by warping the input image, the correspondence map is what the algorithm really produces.

I'm afraid we don't expose the interface for producing this figure. But you can draw things like that by rendering the mesh (vertices + triangulations) represented by various variables in  [generate_profile_faces](https://github.com/hhj1897/face_pose_augmentation/blob/ba26c6845842043f375288588456de17e9341d2d/ibug/face_pose_augmentation/fpa/face_pose_augmentation.py#L16).

**Comment by Mayur28** [comment]
Understood, thank you for your response.
